src/main/java/com/imsweb/layout/naaccrxml/NaaccrXmlLayout.java

```diff
@@ -0,0 +1,467 @@
+     * @param layoutName String used as layout name
+     * @param dictionaries List (of type NaaccrDictionary) of custom user dictionaries - used for making custom layouts
+     * @param loadFields determines whether to load all the fields from all dicitonaries into the allFields variable
+     */
+    public NaaccrXmlLayout(String naaccrVersion, String recordType, String layoutId, String layoutName, List<NaaccrDictionary> dictionaries, boolean loadFields) {
+        _naaccrVersion = naaccrVersion;
+        _layoutName = layoutName;
+        _layoutId = layoutId;
+        _recordType = recordType;
+        _userDictionaries = dictionaries;
+
+        //Only load dictionaries/fields if specified, otherwise avoid expensive operations
+        if (loadFields) {
+            //Load base dictionary from version
+            _baseDictionary = NaaccrXmlDictionaryUtils.getBaseDictionaryByVersion(naaccrVersion);
+
+            //If user dictionaries weren't provided, use default user dictionary
+            if (_userDictionaries == null || _userDictionaries.isEmpty())
+                _userDictionaries = Collections.singletonList(NaaccrXmlDictionaryUtils.getDefaultUserDictionaryByVersion(naaccrVersion));
+
+            //add description from base and user dictionaries
+            StringBuilder description = new StringBuilder("This layout uses the base dictionary version " + _naaccrVersion);
+            description.append(". This layout also uses ").append(_userDictionaries.size()).append(" user dictionaries with URI's ending in: ");
+            for (NaaccrDictionary dictionary : _userDictionaries) {
+                String uri = dictionary.getDictionaryUri();
+                if (uri.length() > 20)
+                    uri = uri.substring(uri.length() - 20);
+                description.append(uri).append(", ");
+            }
+            description.setLength(description.length() - 2);
+            _layoutDesc = description.toString();
+
+            //Load the flat file layout corresponding to the base dictionary and record type for this layout
+            StringBuilder flatFileId = new StringBuilder("naaccr-");
+            flatFileId.append(_naaccrVersion.substring(0, 2)).append("-");
+            switch (_recordType) {
+                case "M":
+                    flatFileId.append("modified");
+                    break;
+                case "C":
+                    flatFileId.append("confidential");
+                    break;
+                case "I":
+                    flatFileId.append("incidence");
+                    break;
+                default:
+                    flatFileId.append("abstract");
+                    break;
+            }
+            _flatLayout = (NaaccrLayout)LayoutFactory.getLayout(flatFileId.toString());
+            if (_flatLayout == null)
+                throw new RuntimeException("Could not find corresponding flat layout.");
```

Comment:
Since you are building the ID, you should print it in your exception so we know what we tried to load. Otherwise it will be much harder to identify the problem.